lib/action_controller/serialization.rb

```diff
@@ -23,7 +23,19 @@ def serialization_scope(scope)
     end
 
     def namespace_for_serializer
-      @namespace_for_serializer ||= self.class.parent unless self.class.parent == Object
+      @namespace_for_serializer ||= namespace_for_class(self.class) unless namespace_for_class(self.class) == Object
+    end
+
+    if Module.method_defined?(:module_parent)
+      def namespace_for_class(klass)
+        klass.module_parent
+      end
+    elsif Module.method_defined?(:parent)
+      def namespace_for_class(klass)
+        klass.parent
+      end
+    else
+      fail "Module must define either parent or module_parent"
```

Comment:
ok, last change by rubocop `fail 'Module must define either parent or module_parent'`

Reply:
sorry - for some reason I didn't see this - just changing double quotes to singles and pushing back up shortly